src/hardware/agilent-54621d/api.c

```diff
@@ -0,0 +1,927 @@
+			*data = std_gvar_tuple_array(*model->vdivs, model->num_vdivs);
+			break;
+		case SR_CONF_LOGIC_THRESHOLD:
+			if (!cg)
+				return SR_ERR_CHANNEL_GROUP;
+			if (!model)
+				return SR_ERR_ARG;
+			*data = g_variant_new_strv(*model->logic_threshold, model->num_logic_threshold);
+			break;
+		case SR_CONF_DATA_SOURCE:
+			*data = g_variant_new_strv(ARRAY_AND_SIZE(data_sources));
+			break;
+	/* TODO Check if all relevant option are present here */
+	default:
+		sr_info("Trying to query for config list for:");
+		return SR_ERR_NA;
+	}
+
+	return ret;
+}
+
+
+static int dev_acquisition_start(const struct sr_dev_inst *sdi)
+{
+	GSList *l;
+	struct sr_channel *ch;
+	struct dev_context *devc;
+	struct sr_scpi_dev_inst *scpi;
+	struct scope_state *state;
+	const struct scope_config *model;
+	int tmp_int;
+	gboolean tmp_bool;
+	char command[MAX_COMMAND_SIZE];
+	float xinc;
+	int points;
+	char *tmp_string;
+	float tmp_float;
+	struct analog_channel_transfer_info *encoding;
+	gboolean digital_added[MAX_DIGITAL_GROUP_COUNT];
+	size_t group, pod_count;
+
+
+	scpi = sdi->conn;
+	devc = sdi->priv;
+	state = devc->model_state;
+	model = devc->model_config;
+
+	devc->num_samples = 0;
+	devc->num_frames = 0;
+
+	//reset to empty
+	for (group = 0; group < ARRAY_SIZE(digital_added); group++)
+		digital_added[group] = FALSE;
+	g_slist_free(devc->enabled_channels);
+	devc->enabled_channels = NULL;
+
+	pod_count = 0;
+	for (l = sdi->channels; l; l = l->next) {
+		ch = l->data;
+		sr_dbg("initializing channel %s", ch->name);
+		if(ch->type == SR_CHANNEL_ANALOG) {
+			if(ch->enabled)
+				devc->enabled_channels = g_slist_append(devc->enabled_channels, ch);
+			//Enable/disable channel if neccessary
+			if(ch->enabled != state->analog_channels[ch->index].state) {
+				if( sr_scpi_send(scpi, (*model->scpi_dialect)[SCPI_CMD_SET_ANALOG_CHAN_STATE], ch->index + 1, ch->enabled ? "ON" : "OFF") != SR_OK || 
+						sr_scpi_get_opc(scpi) != SR_OK)
+					return SR_ERR;
+				state->analog_channels[ch->index].state = ch->enabled;
+			}	
+		} else if (ch->type == SR_CHANNEL_LOGIC) {
+			group = ch->index / DIGITAL_CHANNELS_PER_POD;
+			if(ch->enabled && !digital_added[group]){ 						//Only add a single channel per pod
+				devc->enabled_channels = g_slist_append(devc->enabled_channels, ch);
+				digital_added[group] = TRUE;
+				pod_count++;
+			}									
+			//Enable/disable channel if neccessary
+			if(ch->enabled != state->digital_channels[ch->index]) {
+				if( sr_scpi_send(scpi, (*model->scpi_dialect)[SCPI_CMD_SET_DIG_CHAN_STATE], ch->index, ch->enabled ? "ON" : "OFF") != SR_OK || 
+						sr_scpi_get_opc(scpi) != SR_OK)
+					return SR_ERR;
+				state->digital_channels[ch->index] = ch->enabled;
+			}
+		}	
+	}
+	devc->current_channel = devc->enabled_channels;
+	devc->pod_count = pod_count;
+
+
+	if(devc->data_source == DATA_SOURCE_LIVE){
+		sr_scpi_get_bool(scpi, ":TER?", &tmp_bool);
+		sr_scpi_send(scpi, ":SING");
+		sr_scpi_get_bool(scpi, ":TER?", &tmp_bool);
```

Comment:
The first query resets the "Trigger complete" bit in the devices status register. It is not really of interest if it is set before the next acquisition is stated, however I do sr_scpi_get_bool instead of sr_scpi_send have this be blocking. The agilent scope acts somewhat inconsitently if you continue sending commands while it is still sending the reply to a previous query